### DISCUSSION TITLE
Kiali Traffic Graph Issue: Only Showing Monitoring Namespace Graph

### DISCUSSION THREAD

**Discussion by ganesh1255** [kiali/kiali#7868] [Q&A]
I’ve recently installed Kiali(v1.86.0) on my cluster. My setup includes Istio deployed in the istio-system namespace, and Prometheus, Grafana, and Kiali in the monitoring namespace. Initially, I couldn't see the traffic graph for any namespace in Kiali.

To address this, I added an additional scrape configuration to Prometheus:
prometheus Spec:
  additionalScrapeConfigs:
    - job_name: 'istio'
      metrics_path: '/stats/prometheus'
      kubernetes_sd_configs:
        - role: endpoints
      relabel_configs:
        - source_labels: [meta_kubernetes_service_annotat]
          action: keep
          regex: true

After adding this, I was able to see the traffic graph for the monitoring namespace. However, other namespaces still display only an idle graph, even though I can see results for the istio_requests_total query in the Prometheus UI.

Does anyone have suggestions on how to get Kiali to display traffic graphs for all namespaces?

Thank you in advance!

**Comment by jmazzitelli** [comment]
(_moved this to Q&A Discussions_)

> However, other namespaces still display only an idle graph

For graphs that appear empty, see the FAQ "[Why is my graph empty?](https://kiali.io/docs/faq/graph/#emptygraph)" for details on this to try to debug the problem.

**Reply by ganesh1255** [reply]
Thanks for the documentation @jmazzitelli , tried these queries in prometheus UI
istio_requests_total
istio_tcp_sent_bytes_total
I can see results from all the namespaces but not the graph in kiali

**Reply by jshaughn** [reply]
I assume the namespaces you are interested in are selected in the Namespace dropdown? Perhaps you don't have enough traffic to generate non-zero traffic rates for the time period? Try enabling Show Idle Edges in the Display menu, does anything appear?  Also, just having **some** entries in those Prom metrics doesn't necessarily mean you have traffic in the desired namespaces.  You may need to get more specific with the Prom query like `istio_requests_total{destination_workload_namespace="MyDesiredNamespace"}`

**Reply by ganesh1255** [reply]
Namespaces are selected in the namespace dropdown, i can see idle edges tried specific prom queries, i can see results for desired namespaces in Prometheus UI, but idle graph in Kiali.

**Reply by jshaughn** [reply]
If you see idle edges but not live edges it means that you have no meaningful traffic for the selected time period.  The time period is selected on screen in the duration dropdown.  For example, "Last 2m" would mean the traffic for the from 2 minutes ago to the present time.  A small number of requests may result in a zero rate, after rounding.

**Reply by SID21MAH** [reply]
I am also having same issue, kiali is showing inbound traffic only for observability namespace where kiali operator, Prometheus and Grafana are installed but not for other namespaces, what was resolution for this issue.

**Reply by jshaughn** [reply]
It sounds to me like your sidecars are not generating metrics or prometheus is not scraping them.
